### PR TITLE
Introduce AMReX configuration query object

### PR DESCRIPTION
**Pull request by sayerhs**
Use a Python class with read-only static properties to allow Python query of
AMReX build time configuration.

### CODE REVIEW

File: src/Base/AMReX.cpp [hunk 1]
```diff
@@ -21,6 +25,51 @@ void init_AMReX(py::module& m)
...
+            })
+        .def_property_readonly_static(
+            "have_gpu",
+            [](py::object){
+#ifdef AMREX_USE_GPU
+                return true;
+#else
+                return false;
+#endif
+            })
```
Comment (ax3l) [inline src/Base/AMReX.cpp:57]:
```suggestion
            })
        .def_property_readonly_static(
            "have_omp",
            [](py::object){
#ifdef AMREX_USE_OMP
                return true;
#else
                return false;
#endif
            })
```

**Review by ax3l — COMMENTED** [review]
Nice, thank you!

File: src/Base/AMReX.cpp [hunk 2]
```diff
@@ -21,6 +25,51 @@ void init_AMReX(py::module& m)
                     py::return_value_policy::reference)
         ;
 
+    py::class_<Config>(m, "Config")
```
Comment (ax3l) [inline src/Base/AMReX.cpp:28]: Shall we move this over to a `Config.cpp` file?

Can act as a py-counterpart to `AMReX_Config.H`.

Comment (sayerhs) [reply src/Base/AMReX.cpp:28]: I thought about that... but `Verbose(), OutStream(), Version()` are defined in `AMReX.H` so it felt better to keep it along with other `AMReX.H` functions.

Comment (sayerhs) [reply src/Base/AMReX.cpp:28]: May be we should name that class `Settings` instead if `Config` is confusing... that still doesn't play nice with `OutStream` and `Print` though :(

Also `SpaceDim` is in `AMReX_SPACE.H`

Comment (ax3l) [reply src/Base/AMReX.cpp:28]: Settings is a great idea, too.

I often called this `get(Build)Variant(s)` in the past, but that's probably not too obvious either.

Comment (ax3l) [reply src/Base/AMReX.cpp:28]: Slack: staying with *config* for now since its similar in other python packages (e.g. mpl)

**Review by ax3l — APPROVED** [review]

**Pull request merged**